### ISSUE TITLE
Automated way to import preexisting list via this plugin?

### ISSUE THREAD

**Issue by PaperOrb** [mProjectsCode/obsidian-media-db-plugin#26]
I'm sure many of us already had lists we kept of our movies/games/etc before looking at this plugin. Is there an automated way to import those lists into obsidian using this plugin? It looks great btw, thank you for making this.

**Comment by mProjectsCode** [comment]
@PaperOrb thanks :)

No there currently is no way to bulk import entries. 
I could implement a bulk import by ID, but idk how useful that would be.

**Comment by PaperOrb** [comment]
For my particular use case, I have spreadsheet of media. We can already import each row as a separate note with autogenerated yaml based off of the columns via https://github.com/farling42/obsidian-import-json. People that literally just kept a list of stuff could also easily use external programs to turn it into a CSV to work with farlings plugin.

With that in mind, I think @GarDrastic has the right idea, but with the caveat that it doesnt overwrite preexisting yaml, but rather appends to it. Also if this were to be implemented, I'd recommend setting it up to append media DB YAML to notes in folder rather than just the active note (for instance, I have several hundred notes I have to do this with).

**Comment by mProjectsCode** [comment]
I see what you mean, but this is going to be difficult. If you e.g. have `Star Wars Episode IV` in your list and you want the plugin to find it automatically, then the results are going to be
![image](https://user-images.githubusercontent.com/61602848/172664857-ecfaf2e2-a331-47f6-b09b-2f43eb637622.png)

And none of the options is actually what you want since the movie is named just `Star Wars` on OMDB.

**Comment by PaperOrb** [comment]
Makes sense. Perhaps a semi-automated middle ground could be populating the API search with the title for each media note in a folder so the user doesn't have to type every title in. The plugin could pause on the dropdown part to let the user manually pick the show or give them a chance to redo the search before moving onto the next note?

**Comment by mProjectsCode** [comment]
@PaperOrb 
I implemented your idea. You can test it if you want by cloning the repo and building the plugin yourself. (You need to right-click a folder to open the import modal)
I want to test it some more, before making a release.
Also, it is missing documentation.

**Comment by PaperOrb** [comment]
> @PaperOrb I implemented your idea. You can test it if you want by cloning the repo and building the plugin yourself. (You need to right-click a folder to open the import modal) I want to test it some more, before making a release. Also, it is missing documentation.

I'm amazed you did this and got it out so soon. It seems to be doing great with my anime list right now. A few things I noticed which you probably did as well: 
1. the cancel button just skips to the next show in the folder rather than actually cancelling out of the modal so you can go do something else. had to exit obsidian to get it to stop.
2. entirely dependent on using your mouse right now (hotkeys like tab/enter won't proceed to the next show. you need to click a show title, then scroll down until the ok button appears and click it). Hotkeys could make testing+using this a heck of a lot faster ;)
3. it generates entirely new media files rather than appending to the old ones, meaning your old data would need to be manually merged with the new file (for instance, I have some notes about certain media as well as an ETA field for when the sequel might drop). If you did decide to make it append, files could end up having duplicate yaml fields like the title, so the documentation could mention to check what data the API pulls ahead of time so users change their column names to something unique.

**Comment by mProjectsCode** [comment]
@PaperOrb 

Hey, i did implement a cancel button that cancels the entire import and a skip button that only skips the current note. 
Also, you can now append the original file and it will copy the metadata as well as the file content.
But I don't quite know how to make the modal keyboard accessible.

**Comment by PaperOrb** [comment]
> @PaperOrb
> 
> Hey, i did implement a cancel button that cancels the entire import and a skip button that only skips the current note. Also, you can now append the original file and it will copy the metadata as well as the file content. But I don't quite know how to make the modal keyboard accessible.

Awesome, I'll check it out today or tomorrow. As for the modal hotkeys, I asked over in the plug-dev channel on the discord in case this is something you feel like implementing. From what joethei said, you can use https://marcus.se.net/obsidian-plugin-docs/api/classes/Scope to register hotkeys on modal buttons, and the hotkeys should be unregistered after the modal closes. Here's some examples from his own code on using it: https://github.com/joethei/obsidian-rss/blob/223c2d0941357a6036e0e8bd57010169fb800d66/src/modals/ItemModal.ts#L46

**Comment by mProjectsCode** [comment]
> Here's some examples from his own code on using it: https://github.com/joethei/obsidian-rss/blob/223c2d0941357a6036e0e8bd57010169fb800d66/src/modals/ItemModal.ts#L46

That looks good. I am going to try that tomorrow if I find the time.

**Comment by mProjectsCode** [comment]
I am still having some problems with it but for now, I have the arrow keys to navigate up and down in the list, the right arrow to select that result (I wanted to use space but that also closes the modal for some reason...) and enter to confirm.
Space or escape will cancel everything. 
How does that sound?

**Comment by PaperOrb** [comment]
I'm not through my entire anime backlog yet, but from my testing thus far it looks good. Some small typos in the modal: mata data and meta data should be metadata. 

The hotkeys you setup are great imo. There's one minor issue where if there's too many search results (which has been pretty often in my case), you'll need to mousewheel scroll back to the top to see the topmost result. I'm not sure if there's a way to make the modal default to being scrolled to the very top for each new show or not.

For the documentation, it might be good to mention that users should split their shows up into separate folders for processing--that way if they decide to stop mid-session, they can more easily resume adding shows from where they left off. Otherwise, they'll start from the very beginning of their shows list the next time.

All that said, thanks for this nice addition. It looks fully functional ;)
